metabase/metabase
# Warn users they will remove all content when they remove a db

[Product doc](https://www.notion.so/metabase/Prevent-users-from-accidentally-deleting-all-their-saved-questions-76b21366eb3744ec81b6a0a707571daf)
Closes https://github.com/metabase/metabase/issues/27702

## Changes

Before removing a database if the instance contains user-created content such as questions, models, or segments ask for additional confirmation in a list of checkboxes stating that all of these will be removed with the DB.

<img width="693" alt="Screen Shot 2023-01-25 at 5 44 34 PM" src="https://user-images.githubusercontent.com/14301985/214686238-193c4f1f-1faf-4e9f-9173-d78d8b766be6.png">

<img width="685" alt="Screen Shot 2023-01-25 at 5 44 39 PM" src="https://user-images.githubusercontent.com/14301985/214686251-1b3dc717-09ac-45c6-be59-adfd5881f005.png">

## How to verify

- Admin -> Databases
- Open your sample database and copy the connection string
- Connect two new H2 databases using the connection string: "DB with content" and "Empty DB"
- Create a question and a model using "DB with content" tables
- Return back to the Admin -> Databases
- Delete "Empty DB" and ensure it does not show any checkboxes and asks only for name confirmation
- Delete "DB with content" and ensure it shows checkboxes for confirmation of questions and models removal, and then asks to confirm the DB name
